doc/developer/alembic_on_custom_schema.rst

```diff
@@ -0,0 +1,103 @@
+.. _alembic_on_custom_tables:
+
+Alembic on custom tables
+========================
+
+If you want to manage custom tables (GMF tables; not in the ``main`` nor the ``static`` schema)
+with an alembic you must setup a custom alembic process. Here's an example. Others solutions
+are possible, but do not try to reuse the existing ``alembic.ini``.
+
+The following example use ``docker cp``. You can achieve the same with a volume, but you will
+have to set the owner (root) of copied files.
+
+Init alembic
+------------
+
+In your project, add an ``geoportal/alembic_<project_name>`` folder.
+
+Build and run you project.
+
+In the running container initialize alembic in a folder that have **not** already
+an ``alembic.init`` file, like the /tmp folder. Create also directly the first revision file:
+
+.. code::
+
+  docker-compose exec geoportal mkdir /tmp/new_alembic
+  docker-compose exec geoportal bash -c "cd /tmp/new_alembic; alembic init alembic"
+  # Ignore the warning message about setting the alembic.ini file for now.
+  docker-compose exec geoportal bash -c "cd /tmp/new_alembic; alembic revision -m 'Inital revision'"
+
+Run ``docker-compose geoportal ps`` to know the name of the running geoportal container.
+
+Then copy the generated alembic folder in the ``geoportal`` folder of your project:
+
+.. code::
+
+  PROJECT=<your_project>
+  docker cp <geoportal_container_name>:/tmp/new_alembic geoportal/alembic_${PROJECT}
+
+You can now customize the ``alembic.ini`` and the ``env.py``.
+
+Configure alembic.ini
+---------------------
+
+The main change to apply is to use environments variables to run alembic on the right database.
+To do so, use a ``tmpl`` file:
+
+.. code::
+
+  move geoportal/alembic_${PROJECT}/alembic.ini{,.tmpl}
+  rm geoportal/alembic_${PROJECT}/alembic.ini
+  echo "/geoportal/alembic_${PROJECT}/alembic.ini" >> .gitignore
+
+And inside the ``alembic.ini.tmpl`` set the ``sqlalchemy.url`` to
+``postgresql://${PGUSER}:${PGPASSWORD}@${PGHOST}:${PGPORT}/${PGDATABASE}``.
+
+In this file, you may also directly set a new variable ``version_table_schema`` to set the
+name of the schema inside alembic must create its revision table. See:
```

Comment:
I don't understand this part "inside alembic must create its revision table"
Do we man "inside which alembic must create its revision table" ?

Reply:
yes thanks